src/screens/authentication/UsernameRequest/index.js

```diff
@@ -19,70 +42,66 @@ const SubmitIndicator = ({ submitting }) => (
     />
 )
 
-const Error = ({ error }) => (
-  error && (
-    <Body
-      secondary
-      style={styles.error}
-    >
-      {error}
-    </Body>
-  )
+const SubmitButton = ({ disabled, onPress, error }) => (
+  <PrimaryButton
+    onPress={onPress}
+    style={error ? { marginTop: verticalScale(20) } : { marginTop: verticalScale(40) }}
+    disabled={error ? true : disabled}
+  >
+    <Description white>Continuar</Description>
+  </PrimaryButton>
 )
 
-const SubmitButton = ({ disabled, onPress }) => {
-  if (disabled) {
-    return (
-      <GrayButton>
-        <Description white>Continuar</Description>
-      </GrayButton>
-    )
-  } else {
-    return (
-      <PrimaryButton
-        onPress={onPress}
-      >
-        <Description white>Continuar</Description>
-      </PrimaryButton>
-    )
-  }
-}
-
 const Icon = () => (
-  <>
-    <View style={styles.iconBackground} />
-    <SvgXml
-      width={ICON_WIDTH}
-      height={ICON_HEIGHT}
-      style={styles.icon}
-      xml={usernameIcon}
-    />
-  </>
+  <SvgXml
+    width={ICON_WIDTH}
+    height={ICON_HEIGHT}
+    style={styles.icon}
+    xml={usernameIcon}
+  />
+)
+
+const Suggestions = ({ error, suggestions, onSuggestionSelected }) => (
+  error && (
+    <View style={styles.suggestionContainer}>
+      {suggestions.map((suggestion, index) => {
+        if (index < 4) {
+          return (
+            <Button onPress={() => onSuggestionSelected(suggestion)} key={index}>
+              <SuggestionBox style={(index === 3) ? { borderBottomWidth: scale(0) } : {}}>
```

Comment:
This `3` is actually the last index, I dont think it's necessary to change this but consider using computed values from the structure instead of hard coded values. For example `suggestions.length - 1`